app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/ui/TerminationOverviewScreenScaffold.kt

```diff
@@ -26,144 +24,63 @@ import androidx.compose.runtime.mutableStateOf
 import com.hedvig.android.core.designsystem.theme.HedvigTheme
-import com.hedvig.android.core.ui.appbar.m3.TopAppBar
-import com.hedvig.android.core.ui.appbar.m3.TopAppBarActionType
-import com.hedvig.android.core.ui.card.InsuranceCard
-import com.hedvig.android.core.ui.infocard.VectorInfoCard
-import com.hedvig.android.core.ui.preview.rememberPreviewImageLoader
+import com.hedvig.android.core.ui.appbar.TopAppBarWithBackAndClose
 import com.hedvig.android.core.ui.rememberHedvigDateTimeFormatter
-import com.hedvig.android.core.ui.scaffold.HedvigScaffold
 import hedvig.resources.R
-import kotlinx.collections.immutable.toPersistentList
-import kotlinx.datetime.Clock
 import kotlinx.datetime.LocalDate
-import kotlinx.datetime.TimeZone
 import kotlinx.datetime.toJavaLocalDate
-import kotlinx.datetime.toLocalDateTime
 
 @Composable
-internal fun TerminationOverviewScreenScaffold(
+internal fun TerminationScaffold(
   navigateUp: () -> Unit,
-  topAppBarText: String,
+  closeTerminationFlow: () -> Unit,
   modifier: Modifier = Modifier,
   content: @Composable ColumnScope.() -> Unit,
 ) {
-  Surface(color = MaterialTheme.colorScheme.background, modifier = modifier.fillMaxSize()) {
+  Surface(
+    color = MaterialTheme.colorScheme.background,
+    modifier = modifier.fillMaxSize(),
+  ) {
     Column {
-      val scrollBehavior = TopAppBarDefaults.pinnedScrollBehavior()
-      TopAppBar(
-        title = topAppBarText,
-        onClick = navigateUp,
-        actionType = TopAppBarActionType.BACK,
-        colors = TopAppBarDefaults.topAppBarColors(
-          containerColor = MaterialTheme.colorScheme.background,
-          scrolledContainerColor = MaterialTheme.colorScheme.surface,
-        ),
-        scrollBehavior = scrollBehavior,
+      val topAppBarScrollBehavior = TopAppBarDefaults.pinnedScrollBehavior()
+      TopAppBarWithBackAndClose(
+        onNavigateUp = navigateUp,
+        onClose = closeTerminationFlow,
+        title = "",
+        scrollBehavior = topAppBarScrollBehavior,
       )
       Column(
         modifier = Modifier
           .fillMaxSize()
-          .nestedScroll(scrollBehavior.nestedScrollConnection)
+          .nestedScroll(topAppBarScrollBehavior.nestedScrollConnection)
           .verticalScroll(rememberScrollState())
-          .windowInsetsPadding(WindowInsets.safeDrawing.only(WindowInsetsSides.Horizontal)),
+          .windowInsetsPadding(
+            WindowInsets.safeDrawing.only(
+              WindowInsetsSides.Horizontal + WindowInsetsSides.Bottom,
+            ),
+          ),
       ) {
+        Spacer(modifier = Modifier.height(8.dp))
+        Text(
+          text = stringResource(id = R.string.TERMINATION_FLOW_CANCELLATION_TITLE),
+          fontSize = MaterialTheme.typography.headlineSmall.fontSize,
+          fontStyle = MaterialTheme.typography.headlineSmall.fontStyle,
+          fontFamily = MaterialTheme.typography.headlineSmall.fontFamily,
```

Comment:
```suggestion
          style = MaterialTheme.typography.headlineSmall,
```
I **think** this can be replaced just by this, can you give it a try?